uart: unused warning

Since #2701, if UART does not support the rx, tx and data invert warnings are raised about unused variables:
Example with STM32F4 target:

```
.arduino15/packages/STMicroelectronics/hardware/stm32/2.10.1/libraries/SrcWrapper/src/stm32/uart.c: In function 'uart_init':
.arduino15/packages/STMicroelectronics/hardware/stm32/2.10.1/libraries/SrcWrapper/src/stm32/uart.c:118:110: warning: unused parameter 'rx_invert' [-Wunused-parameter]
  118 | bool uart_init(serial_t *obj, uint32_t baudrate, uint32_t databits, uint32_t parity, uint32_t stopbits, bool rx_invert, bool tx_invert, bool data_invert)
      |                                                                                                              ^
.arduino15/packages/STMicroelectronics/hardware/stm32/2.10.1/libraries/SrcWrapper/src/stm32/uart.c:118:126: warning: unused parameter 'tx_invert' [-Wunused-parameter]
  118 | bool uart_init(serial_t *obj, uint32_t baudrate, uint32_t databits, uint32_t parity, uint32_t stopbits, bool rx_invert, bool tx_invert, bool data_invert)
      |                                                                                                                              ^
.arduino15/packages/STMicroelectronics/hardware/stm32/2.10.1/libraries/SrcWrapper/src/stm32/uart.c:118:142: warning: unused parameter 'data_invert' [-Wunused-parameter]
  118 | bool uart_init(serial_t *obj, uint32_t baudrate, uint32_t databits, uint32_t parity, uint32_t stopbits, bool rx_invert, bool tx_invert, bool data_invert)
      | 
```

--- a/libraries/SrcWrapper/src/stm32/uart.c
+++ b/libraries/SrcWrapper/src/stm32/uart.c
@@ -438,6 +438,10 @@ bool uart_init(serial_t *obj, uint32_t baudrate, uint32_t databits, uint32_t par
     huart->AdvancedInit.DataInvert = UART_ADVFEATURE_DATAINV_ENABLE;
   }
 #endif
+#else /* UART_ADVFEATURE_NO_INIT */
+  UNUSED(rx_invert);
+  UNUSED(tx_invert);
+  UNUSED(data_invert);
 #endif
 #ifdef UART_ONE_BIT_SAMPLE_DISABLE
   huart->Init.OneBitSampling = UART_ONE_BIT_SAMPLE_DISABLE;

